### PR TITLE
Function structure refactor

### PR DESCRIPTION
**Pull request by TechDufus**

### CODE REVIEW

File: commands/weather.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            5: ':cloud_rain: :frowning:',
+            6: ':cloud_snow: :snowflake: :snowman:',
+            7: ':foggy: :eyeglasses:',
+            8: ':white_sun_small_cloud: :sunny: :fire:'
+        }
+        emojistr = emojistr.get(code, '')
+    else:
+        emojistr = ':cloud: :slight_frown: :cloud: :slight_frown: :cloud:'
+
+    await ctx.send(f"**The Weather for {station} is:** {forecast} {emojistr}")
```
Comment (xdega) [inline commands/weather.py:31]: This is missing a newline at the end of the file
```suggestion
    await ctx.send(f"**The Weather for {station} is:** {forecast} {emojistr}")
    
```

File: commands/roll.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+async def roll(ctx):
+    """ Rolls a pseudorandom number from 0 to 100 """
```
Comment (xdega) [inline commands/roll.py:2]: Lets update the comment syntax here:
```suggestion
    # Rolls a pseudorandom number from 0 to 100
```

File: commands/roll.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    num = randint(0, 100)
+    num = str(num)
```
Comment (xdega) [inline commands/roll.py:4]: These two lines can probably be combined:
```suggestion
    num = str(randint(0, 100))
```

File: commands/roll.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+    await ctx.send(f"{ctx.message.author.mention} rolled {num}")
```
Comment (xdega) [inline commands/roll.py:5]: Newline at end of file
```suggestion
    await ctx.send(f"{ctx.message.author.mention} rolled {num}")
    
```

File: commands/weather.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+async def weather(ctx, zipcode):
+    """ Gets the current weather for given US zip code """
```
Comment (xdega) [inline commands/weather.py:2]: Comment syntax
```suggestion
    # Gets the current weather for given US zip code
```

**Comment by TechDufus** [comment]
I've reviewed this, but can't test that this new import syntax and commands still work until testing this in staging.

File: commands/weather.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    url = 'https://api.openweathermap.org/data/2.5/weather?'
+    url += f"zip={zipcode}"
+    url += f"&appid={WEATHER_API_KEY}"
```
Comment (xdega) [inline commands/weather.py:5]: It probably makes more sense to combine these lines, now that we are using f strings:
```suggestion
    url += f"zip={zipcode}&appid={WEATHER_API_KEY}"
```

File: commands/nasa_apod.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    url = f"https://api.nasa.gov/planetary/apod?api_key={NASA_API_KEY}"
+    response = requests.get(url)
+    response = response.json()
+
+    print(response)
+
+    photo = response['url']
+    title = response['title']
+
+    await ctx.send(f"**Description:** {title}\n{photo}")
```
Comment (xdega) [inline commands/nasa_apod.py:12]: Newline at end of file
```suggestion
    await ctx.send(f"**Description:** {title}\n{photo}")
    
```

File: commands/nasa_apod.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+async def nasa_apod(ctx):
+    """ Sends a beautiful space picture of the day from NASA """
```
Comment (xdega) [inline commands/nasa_apod.py:2]: Comment syntax
```suggestion
    # Sends a beautiful space picture of the day from NASA
```

File: commands/nasa_apod.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    url = f"https://api.nasa.gov/planetary/apod?api_key={NASA_API_KEY}"
+    response = requests.get(url)
+    response = response.json()
```
Comment (xdega) [inline commands/nasa_apod.py:5]: This doesn't have to be addressed in this update, but we will probably want to consolidate these two lines. Perhaps even factor out into a "request" helper function that can be shared between commands, as this is commonly used.

File: commands/help.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+async def help(ctx):
+    """ Sends help message to the user, informing of available commands """
+    message = """
+Hey there! You need help?
+Here are the commands I respond to:
+** help ** - I will whisper you this help message.
+** roll ** - Nothing unique. I'll just roll a dice from 1-100. Good Luck!
+** weather [zip code] ** - Let you know the weather for zipcode, so you don't have to go outside.
+** nasa_apod ** - Displays the Astral Photo of the Day, provided by the NASA API.
```
Comment (xdega) [inline commands/help.py:9]: The description here is a bit off from the official documentation.
```suggestion
** nasa_apod ** - Displays the Astronomy Picture of the Day, provided by the NASA API.
```

File: commands/help.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+    """
+    await ctx.send(message)
```
Comment (xdega) [inline commands/help.py:11]: Newline at end of file:
```suggestion
    await ctx.send(message)
    
```

File: commands/help.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+async def help(ctx):
+    """ Sends help message to the user, informing of available commands """
```
Comment (xdega) [inline commands/help.py:2]: Comment syntax:

```suggestion
    # Sends help message to the user, informing of available commands
```

File: bot.py [hunk 13]
```diff
@@ -8,102 +8,30 @@
...
-
-    await ctx.send('**The Weather for ' + station + ' is: **' + forecast + ' ' + emojistr)
-
-@bot.command(pass_context=True)
-async def roll(ctx):
-    """ Rolls a pseudorandom number from 0 to 100 """
-    num = randint(0, 100)
-    num = str(num)
-    await ctx.send(ctx.message.author.mention + ' rolled ' + num)
+# commands
```
Comment (xdega) [inline bot.py:32]:
```suggestion
# Commands
```

File: bot.py [hunk 14]
```diff
@@ -8,102 +8,30 @@
...
+for file in os.listdir('commands'):
+    @bot.command(pass_context=True)
+    exec(f"commands/{file}")
```
Comment (xdega) [inline bot.py:35]: Did you ever get the syntax issue resolved here? Or are we going to be merging this in a broken state?

File: bot.py [hunk 15]
```diff
@@ -8,102 +8,30 @@
 import discord
 from discord.ext import commands
 
+
 BOT_PREFIX = os.environ['discord_prefix']
 TOKEN = os.environ['discord_token']
 WEATHER_API_KEY = os.environ['open_weather_map_key']
 NASA_API_KEY = os.environ['nasa_api_key']
```
Comment (xdega) [inline bot.py:15]: These environment variables could probably be defined in the function files for the commands.

File: commands/weather.py [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+
+    if int(code) <= 800:
+        code = int(str(code)[:1])
+        emojistr = {
+            2: ':thunder_cloud_rain: :fearful:',
+            3: ':white_sun_rain_cloud:',
+            5: ':cloud_rain: :frowning:',
+            6: ':cloud_snow: :snowflake: :snowman:',
+            7: ':foggy: :eyeglasses:',
+            8: ':white_sun_small_cloud: :sunny: :fire:'
```
Comment (xdega) [inline commands/weather.py:25]: Lets use double quotes across the board for the sake of consistency.

```suggestion
            2: ":thunder_cloud_rain: :fearful:",
            3: ":white_sun_rain_cloud:",
            5: ":cloud_rain: :frowning:",
            6: ":cloud_snow: :snowflake: :snowman:",
            7: ":foggy: :eyeglasses:",
            8: ":white_sun_small_cloud: :sunny: :fire:"
```

File: commands/weather.py [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        emojistr = emojistr.get(code, '')
+    else:
+        emojistr = ':cloud: :slight_frown: :cloud: :slight_frown: :cloud:'
```
Comment (xdega) [inline commands/weather.py:29]:
```suggestion
        emojistr = ":cloud: :slight_frown: :cloud: :slight_frown: :cloud:"
```

**Review by xdega — CHANGES_REQUESTED** [review]

File: commands/nasa_apod.py (hunk 9 above)
Comment (TechDufus) [reply commands/nasa_apod.py:5]: Can create a new package directory for helper functions for this.

**Pull request merged**